### PR TITLE
[Issue #177] feat: add logic to sync transactions after login

### PR DESCRIPTION
**Pull request by chedieck**
**Description:**
Part of #177 
Added signin / signup callbacks, as described in [the supertokens docs](https://supertokens.com/docs/thirdpartyemailpassword/common-customizations/handling-signinup-success).
Post login now triggers the syncing of the transactions related to each one of the logged in user's addresses.

**Test plan:**
With the BCH / XEC nodes URLS defined in the `.env` file such as:
```
GRPC_BCH_NODE_URL=bchd.fountainhead.cash:443    
GRPC_XEC_NODE_URL=grpc.fabien.cash:8335         
```
...do:
1. `make reset-dev`
2. Log in into the default dev user on http://localhost:3000
3. Go to buttons, select any of them. All their transactions should appear on the button detail page (the syncing of the transactions may take some seconds, from the moment of the login, to happen).

### CODE REVIEW

**Review by Klakurka — CHANGES_REQUESTED** [review]
ecash:qrmm7edwuj4jf7tnvygjyztyy0a0qxvl7quss2vxek doesn't seem to sync for me (no txs showing)

I also can't run the syncs manually, getting a 404.

**Comment by chedieck** [comment]
> ecash:qrmm7edwuj4jf7tnvygjyztyy0a0qxvl7quss2vxek doesn't seem to sync for me (no txs showing)

I think this is that same bug from before on the XEC GRPC client; I have confirmed that the problem is **not** on the node and am working on fixing it.

> I also can't run the syncs manually, getting a 404.

Indeed I removed the sync endpoint.
Independently from the bug aforementioned, should the post-login hook parse if the sync was indeed successful and then do something about it? This would probably be a separate task but it will be good to think about how to parse it.

**Review by Klakurka — CHANGES_REQUESTED** [review]
It definitely at least works in my testing but I'll leave the technical review / pondering alternatives to better men.

Unit tests are failing though.

**Comment by chedieck** [comment]
Fixed the unittests, also removed `rewire` for good (it was causing more problems than it was solving).

**Review by Klakurka — APPROVED** [review]

**Comment by chedieck** [comment]
> @chedieck please look into #208 first as it will remove the need for these spyOn lines I believe.

Indeed, I just tried merging this branch into #208 and there was this single conflict: by selecting your new changes (thus removing all the `spyOn` code) the tests still all passed and the sync still worked.

File: config/backendConfig.ts [hunk 1]
```diff
@@ -64,7 +66,70 @@ export const backendConfig = (): TypeInput => {
...
+              // override the email password sign up API
+              emailPasswordSignUpPOST: async function (input) {
+                if (originalImplementation.emailPasswordSignUpPOST === undefined) {
+                  throw Error('Should never come here')
+                }
+                // pre sign up logic goes here
+
+                const response = await originalImplementation.emailPasswordSignUpPOST(input)
+
+                if (response.status === 'OK') {
```
Comment (Gattaca1) [inline config/backendConfig.ts:84]: Empty If statement here, any good reason to leave it in?

**Review by Gattaca1 — CHANGES_REQUESTED** [review]

File: config/backendConfig.ts (hunk 1 above)
Comment (chedieck) [reply config/backendConfig.ts:84]: > Empty If statement here, any good reason to leave it in?

I think so, this was basically copied and pasted from their docs, so it's kind of a template on which we can later implement pre/post login hooks.

Comment (Klakurka) [reply config/backendConfig.ts:84]: Sounds reasonable to me.

**Review by Klakurka — APPROVED** [review]

**Pull request merged**